cocos/core/data/garbage-collection.ts

```diff
@@ -27,33 +27,53 @@ declare class FinalizationRegistry {
     unregister (token: any);
 }
 
+const targetSymbol = Symbol('[[target]]');
+
 class GarbageCollectionManager {
     private _finalizationRegistry: FinalizationRegistry | null = EDITOR ? new FinalizationRegistry(this.finalizationRegistryCallback.bind(this)) : null;
+    private _gcObjects: WeakMap<any, GCObject> = new WeakMap();
 
     public registerGCObject (gcObject: GCObject): GCObject {
         if (EDITOR) {
-            gcObject._finalizationToken = {};
-            const proxy = new Proxy(gcObject, {});
-            this._finalizationRegistry!.register(proxy, gcObject, gcObject._finalizationToken);
+            const token = {};
+            const proxy = new Proxy(gcObject, {
+                get (target, property, receiver) {
+                    if (property === targetSymbol) {
+                        return target;
+                    }
+                    let val = target[property];
+                    if (typeof val === 'function' && property !== 'constructor') {
+                        const original = val;
+                        val = function newFunc () {
+                            // @ts-expect-error this is referenced to proxy
+                            return original.apply(this[targetSymbol], arguments) as unknown;
+                        };
+                    }
+                    return val as unknown;
+                },
+                set (target, prop, value, receiver) {
+                    target[prop] = value;
+                    return true;
+                },
+            });
+            this._gcObjects.set(token, gcObject);
+            this._finalizationRegistry!.register(proxy, token, token);
```

Comment:
Is this overwriting all functions of the target gc object ? If so, it could hurt performance of the object and of the invocation process. But I'm not sure what the register do.

Reply:
FinalizationRegistry has no impact on performance, Proxy has a larger impact on performance, so it is intended to be used only in the editor

Reply:
As long as it only affects the editor environment, I'm ok